In [332]:
import pandas as pd

In [333]:
df=pd.read_excel("Challenge_407_start_file/Draws_Lottery.xlsx")

In [334]:
newdata=df

In [335]:
newdata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2092 entries, 0 to 2091
Data columns (total 16 columns):
 #   Column                                Non-Null Count  Dtype 
---  ------                                --------------  ----- 
 0   Draw ID                               2092 non-null   int64 
 1   Draw Date                             2092 non-null   object
 2   Winning number1                       2092 non-null   int64 
 3   Winning number2                       2092 non-null   int64 
 4   Winning number3                       2092 non-null   int64 
 5   Winning number4                       2092 non-null   int64 
 6   Winning number5                       2092 non-null   int64 
 7   Winning number6                       2092 non-null   int64 
 8   Matches 6/6                           2092 non-null   int64 
 9   Prize won by each player (match 6/6)  2092 non-null   object
 10  Matches 5/6                           2092 non-null   int64 
 11  Prize won by each player (matc

**To find:**

 just one winner (matched 6 out of 6 numbers) who claimed the entire accumulated jackpot

In [336]:
newdata=newdata[newdata["Matches 6/6"]==1]
newdata.count()["Draw ID"]

364

There were **364 Winners** (matched 6 out of 6 numbers) who claimed the entire accumulated jackpot

Now to Identify the **date** and the **highest jackpot amount** ever won by a single winner who matched 6 out of 6 numbers

In [337]:
newdata["Prize won by each player (match 6/6)"]
##we have to find the max of this value. Since this is a string data type this has to be converted to Float

1        2.307.162,23
10      15.591.365,07
16       6.789.869,08
23      18.661.679,61
28       5.401.793,60
            ...      
2080    21.763.750,54
2081     2.420.701,10
2086    27.364.769,44
2087     2.526.635,00
2090    20.024.894,84
Name: Prize won by each player (match 6/6), Length: 364, dtype: object

In [338]:
newdata[["before","after"]]=newdata["Prize won by each player (match 6/6)"].str.split(",",expand=True)
newdata[["before","after"]]
#splitting the data

c:\Users\divya\anaconda3\lib\site-packages\pandas\core\frame.py:3641: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[k1] = value[k2]


before after
1      2.307.162    23
10    15.591.365    07
16     6.789.869    08
23    18.661.679    61
28     5.401.793    60
...          ...   ...
2080  21.763.750    54
2081   2.420.701    10
2086  27.364.769    44
2087   2.526.635    00
2090  20.024.894    84

[364 rows x 2 columns]

In [339]:
newdata["before"]=newdata["before"].str.replace(".","")
#removing the points inbetween to make it typ float

C:\Users\divya\AppData\Local\Temp/ipykernel_22260/3912435223.py:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  newdata["before"]=newdata["before"].str.replace(".","")
C:\Users\divya\AppData\Local\Temp/ipykernel_22260/3912435223.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  newdata["before"]=newdata["before"].str.replace(".","")


In [340]:
newdata["Prize won by each player (match 6/6)"]=newdata["before"]+"."+newdata["after"]
newdata["Prize won by each player (match 6/6)"]
#combining the two split columns to get proper number

C:\Users\divya\AppData\Local\Temp/ipykernel_22260/3211085667.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  newdata["Prize won by each player (match 6/6)"]=newdata["before"]+"."+newdata["after"]


1        2307162.23
10      15591365.07
16       6789869.08
23      18661679.61
28       5401793.60
           ...     
2080    21763750.54
2081     2420701.10
2086    27364769.44
2087     2526635.00
2090    20024894.84
Name: Prize won by each player (match 6/6), Length: 364, dtype: object

In [341]:
newdata["Prize won by each player (match 6/6)"]=newdata["Prize won by each player (match 6/6)"].astype(float)
#changing the data type to float

C:\Users\divya\AppData\Local\Temp/ipykernel_22260/3496673918.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  newdata["Prize won by each player (match 6/6)"]=newdata["Prize won by each player (match 6/6)"].astype(float)


In [342]:
newdata["Prize won by each player (match 6/6)"].max()

205329753.89

In [346]:
pd.options.display.float_format = '{:.2f}'.format
newdata[newdata["Prize won by each player (match 6/6)"]==newdata["Prize won by each player (match 6/6)"].max()][["Draw Date","Prize won by each player (match 6/6)"]]
# we want only the date and amount of highest jackpot amount

Draw Date  Prize won by each player (match 6/6)
1763  25/11/2015                          205329753.89

So the **Higest Jackpot Won** was of amount ***205,329,753.89*** on ***25/11/2015***